com.unity.multiplayer.mlapi/Tests/Runtime/NetworkObject/NetworkObjectDestroyTests.cs

```diff
@@ -0,0 +1,245 @@
+using System.Collections;
+using MLAPI.Configuration;
+using NUnit.Framework;
+using UnityEngine;
+using UnityEngine.TestTools;
+
+namespace MLAPI.RuntimeTests
+{
+    public class NetworkObjectDestroyTests
+    {
+        private const int k_ClientInstanceCount = 2;
+
+        private NetworkManager m_ServerNetworkManager;
+        private NetworkManager[] m_ClientNetworkManagers;
+        private GameObject m_PlayerPrefab;
+        private GameObject m_DummyPrefab;
+        private GameObject m_DummyGameObject;
+
+        [UnitySetUp]
+        public IEnumerator Setup()
+        {
+            // we need at least 1 client for tests
+            Assert.That(k_ClientInstanceCount, Is.GreaterThan(0));
+
+            Assert.That(MultiInstanceHelpers.Create(k_ClientInstanceCount, out m_ServerNetworkManager, out m_ClientNetworkManagers));
+            Assert.That(m_ServerNetworkManager, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers, Is.Not.Null);
+            Assert.That(m_ClientNetworkManagers.Length, Is.EqualTo(k_ClientInstanceCount));
+
+            m_PlayerPrefab = new GameObject("PlayerPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_PlayerPrefab.AddComponent<NetworkObject>());
+
+            m_DummyPrefab = new GameObject("DummyPrefabPrototype");
+            MultiInstanceHelpers.MakeNetworkedObjectTestPrefab(m_DummyPrefab.AddComponent<NetworkObject>());
+
+            m_ServerNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+            m_ServerNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            foreach (var clientNetworkManager in m_ClientNetworkManagers)
+            {
+                clientNetworkManager.NetworkConfig.PlayerPrefab = m_PlayerPrefab;
+                clientNetworkManager.NetworkConfig.NetworkPrefabs.Add(new NetworkPrefab { Prefab = m_DummyPrefab });
+            }
+
+            Assert.That(MultiInstanceHelpers.Start(/* isHost = */ true, m_ServerNetworkManager, m_ClientNetworkManagers));
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientsConnected(m_ClientNetworkManagers));
+            yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientConnectedToServer(m_ServerNetworkManager));
+        }
+
+        [TearDown]
+        public void Teardown()
+        {
+            MultiInstanceHelpers.Destroy();
```

Comment:
even though we're calling this method, it looks like the very next unit test holds onto line 45 above:

```
yield return MultiInstanceHelpers.Run(MultiInstanceHelpers.WaitForClientsConnected(m_ClientNetworkManagers));
```

I'm not entirely sure why but this `Setup`/`Teardown` looks legit to me, any thoughts? @TwoTenPvP @NoelStephensUnity